### PR TITLE
feat(web): editor formatting options

### LINKED ISSUE #288: web: message content should highlight links
Links appear inline without any stylistic changes
we should highlight them

### PR DESCRIPTION
**Pull request by mooyg**
## What does this PR do?

Adds more markdown features support to the message box.
* Underline
* Superscript
* Subscript
* Link highlight and opening on click

Fixes #288

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Required

- [x] Read [Contributing Guide](https://github.com/un/inbox/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Tested my code in a local environment
- [x] Commented on my code in hard-to-understand areas
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the UnInbox Docs if changes were necessary

https://github.com/un/inbox/assets/61348580/b8e0f27e-25f8-4951-8770-770f0225b010

### CODE REVIEW

File: packages/tiptap/extensions.ts [hunk 1]
```diff
@@ -15,7 +18,14 @@ export const tipTapExtensions: AnyExtension[] = [
   }),
   Link.configure({
     linkOnPaste: true,
-    autolink: true
+    autolink: true,
+    openOnClick: true,
+    HTMLAttributes: {
+      class: 'href-link'
```
Comment (BlankParticle) [inline packages/tiptap/extensions.ts:24]: you can just use tailwind classnames here, and remove it from global css

Comment (mooyg) [reply packages/tiptap/extensions.ts:24]: This happens at runtime and not build time

Comment (BlankParticle) [reply packages/tiptap/extensions.ts:24]: still should be fine, we have been doing it like always,
If it creates any issues in staging we can always revert it.

Comment (BlankParticle) [reply packages/tiptap/extensions.ts:24]: also don't use `text-blue-400`, we have disabled those colors
use `text-blue-${1..12}` from radix color pallete

**Review by BlankParticle — APPROVED** [review]

**Comment by BlankParticle** [comment]
@mooyg, Please sign the CLA so that I can merge this PR

**Review by McPizza0 — APPROVED** [review]
lgtm (looks good to me)

**Pull request merged**